src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -408,14 +467,44 @@ private void CreateNewIndex(Directory dir)
                     //unlock it!
                     IndexWriter.Unlock(dir);
                 }
+
                 //create the writer (this will overwrite old index files)
                 writer = new IndexWriter(dir, FieldAnalyzer, true, IndexWriter.MaxFieldLength.UNLIMITED);
+                if (_directory is ExamineDirectory examineDirectory)
+                {
```

Comment:
Is it neccessary to call the ExamineDirectory methods here? As far as I can tell all this method overall does is create an indexwriter then disposes it.

Reply:
@nzdev it is not, but I had some weird issues when I didn't called that there, like nomerge policy wasnt working etc.

Reply:
If the delete policy is set, will it still delete the old index files? This methods responsibility appears to be just to delete any old files that exist.

Reply:
They will be not removed, but you can't remove them as it will cause sync issues, this is why Lucene on loadbalanced environment is not a good thing, elastic and solar handle that internally by doing only one source of true and later using aliasing (something similar to that what I have done in readonly), but they do also other tricks and issue is if we will be removing files if sync will be overriden then index will go down, it is why nodeletionpolicy is required to make that work :)

Reply:
Thanks @bielu . Can you please point out in the code where you are doing this aliasing?

Reply:
Also @bielu have you tried this out on a load balanced enviroment?

Reply:
@nzdev I have loadbalanced enviroment setup, which I am using to test all changes, about where I am doing aliasing it is there:
https://github.com/Shazwazza/Examine/pull/197/files#diff-960cf039cc4d5d906becce9df45b53e328312b769876c1445e7dfb360ae3608bR70
When we rebuild cache directory, we still serve old one, as soon we finish rebuild we swap them :) that operation is called aliasing of indexes :)[using name of index (alias) to communicate, but real name of indexes is different]